lms/static/js/utils/dropdown.js

```diff
@@ -0,0 +1,201 @@
+var edx = edx || {},
+    host = window.location.host.substring(0, window.location.host.indexOf(':')),
+    Dropdown = (function() {
+
+        var dropdown = {
+
+            opts: {
+                page: $(document),
+                midpoint: ($(window).width() / 2),
+                button: $('.button-more.has-dropdown'),
+                button_active: 'is-active',
+                menu: $('.dropdown-menu'),
+                menu_inactive: 'is-hidden',
+                menu_active: 'is-visible',
+                menu_align: 'align-',
+                text_align: 'control-text-align'
+            },
+
+            init: function(parent) {
+                if (dropdown.opts.button.length && dropdown.opts.menu.length) {
+
+                    if (parent) {
+                        dropdown.opts.page = $(parent);
+                    }
+
+                    dropdown.listenForClick();
+                    dropdown.listenForKeypress();
+                }
+            },
+
+            optsReset: function() {
+                dropdown.opts = {
```

Comment:
This dict is exactly the same as the one defined on line 8. Don't copy code-- refactor it so it is only defined in one place or helper method.